RuntimeError: dry-run parsing failed

Trying to bisect anything leads to this exception for me. I packaged nix-bisect 0.4.1 into nixpkgs master today but I fail to use it.

```python
❯ extra-bisect start HEAD 7c1912fb9a7372337c5b324fa9f14d58c3ad115f
Bisecting: 128 revisions left to test after this (roughly 7 steps)
[21adb1d7fba7927c99190283bcd6b406f3e34359] python3Packages.spacy-transformers: 1.1.4 -> 1.1.5
❯ extra-bisect run nix-build-status --failure-line "FAILED tests/generic/test_finalize.py::test_binops" python3.pkgs.pandas
$ bisect-env nix-build-status --failure-line 'FAILED tests/generic/test_finalize.py::test_binops' python3.pkgs.pandas
Querying status of /nix/store/l0yan2cpa1bxrblkchnshzw9knryq7hw-python3.9-pandas-1.4.1.drv.
Traceback (most recent call last):
  File "/nix/store/ll3va8aai7la6iyl8y5wjbc2l1vr2y67-nix-bisect-0.4.1/bin/.nix-build-status-wrapped", line 9, in <module>
    sys.exit(_main())
  File "/nix/store/ll3va8aai7la6iyl8y5wjbc2l1vr2y67-nix-bisect-0.4.1/lib/python3.9/site-packages/nix_bisect/build_status.py", line 184, in _main
    status = build_status(
  File "/nix/store/ll3va8aai7la6iyl8y5wjbc2l1vr2y67-nix-bisect-0.4.1/lib/python3.9/site-packages/nix_bisect/build_status.py", line 49, in build_status
    if not drv.can_build_deps():
  File "/nix/store/ll3va8aai7la6iyl8y5wjbc2l1vr2y67-nix-bisect-0.4.1/lib/python3.9/site-packages/nix_bisect/derivation.py", line 39, in can_build_deps
    return nix.build_would_succeed(
  File "/nix/store/ll3va8aai7la6iyl8y5wjbc2l1vr2y67-nix-bisect-0.4.1/lib/python3.9/site-packages/nix_bisect/nix.py", line 287, in build_would_succeed
    rebuilds = build_dry(drvs, nix_options=nix_options)[0]
  File "/nix/store/ll3va8aai7la6iyl8y5wjbc2l1vr2y67-nix-bisect-0.4.1/lib/python3.9/site-packages/nix_bisect/nix.py", line 69, in build_dry
    raise RuntimeError("dry-run parsing failed")
RuntimeError: dry-run parsing failed
```

The derivation looks like this:
```
Derive([("out","/nix/store/5534m7zgrb8jj177aw37cg3na48f4j4a-python3.9-pandas-1.4.1","","")],[("/nix/store/235fdj7lgk6qwdg3m7ygfr681nkqd1s3-python3-3.9.11.drv",["out"]),("/nix/store/3hik582k85ahjw6dm3pc4q0ljh3ys1x7-bash-5.1-p16.drv",["out"]),("/nix/store/42hjr5sgp9gl81kdm86xwgcmr3zzphd6-pandas-1.4.1.tar.gz.drv",["out"]),("/nix/store/50gkwvrbya4chndd7fk0j0shpkqqig8i-pytest-check-hook.drv",["out"]),("/nix/store/5gpq1zrcklxmmrdhjbrpnhrib77l8bfx-python3.9-pytest-asyncio-0.18.1.drv",["out"]),("/nix/store/70fkmkjlmgppgg31zjr90mly835w8hw6-glibc-locales-2.34-115.drv",["out"]),("/nix/store/80dyisjzjzl1ca5750ainc9pn591pnby-python3.9-setuptools-61.2.0.drv",["out"]),("/nix/store/9zy1s0dfazf0pjiqkgzlk8azdh3dr7r2-setuptools-setup-hook.drv",["out"]),("/nix/store/abjmb58ad86hvskawd7zf0jr334sjcgf-python3.9-Cython-0.29.28.drv",["out"]),("/nix/store/d06yx5gnra5i4vkmpf6f66m0bxgfzpab-python-remove-bin-bytecode-hook.drv",["out"]),("/nix/store/ifyrasyy2b1q9f1wjzd381cq76h7raim-python-imports-check-hook.sh.drv",["out"]),("/nix/store/iv6859irlk0jkwsg445g68zx8yn47x5h-python3.9-Jinja2-3.1.1.drv",["out"]),("/nix/store/j0n77b0x212hfzdadx644hkcsa2h1y5n-python-namespaces-hook.sh.drv",["out"]),("/nix/store/j9x9649dddyp9l5gy9pz5hyyq76j9fym-setuptools-check-hook.drv",["out"]),("/nix/store/k0qzigdqi6za0yly8i105991nk9vjhha-hook.drv",["out"]),("/nix/store/k12av414spx4ykvj873l3yg008in392s-python3.9-hypothesis-6.40.0.drv",["out"]),("/nix/store/kbw0mvsg4s9cy72xs41wac02f1whjg06-python3.9-numpy-1.21.5.drv",["out"]),("/nix/store/lb1fqdz918040d7644py17ay3608w5h7-python3.9-pytz-2022.1.drv",["out"]),("/nix/store/mb6nf964hx76nxpcb4i6px3pfjddccyq-python-remove-tests-dir-hook.drv",["out"]),("/nix/store/pgzp9hxjl7dj40hgrl8bgy0pwx32rj2c-pip-install-hook.drv",["out"]),("/nix/store/q09r0mndfdnlgq1nhw5m9c5a3h7l03q8-stdenv-linux.drv",["out"]),("/nix/store/qvdlfwxmbmacinrfjk01r254scvqbpzs-python3.9-xlsxwriter-3.0.2.drv",["out"]),("/nix/store/v0l39l23r48ijzagbx4kq23nc54vk034-python3.9-pytest-xdist-2.5.0.drv",["out"]),("/nix/store/w51r49habzklp3x8vxpzz7rs3vjax1g5-hook.drv",["out"]),("/nix/store/xlfy2q558ci6nfbfyx3qbsxmkz66zl57-python-catch-conflicts-hook.drv",["out"]),("/nix/store/y9p8djm4ximcfcgm87l6q4qky8fz8ylb-python3.9-python-dateutil-2.8.2.drv",["out"])],["/nix/store/9krlzvny65gdc8s7kpb6lkx8cd02c25b-default-builder.sh"],"x86_64-linux","/nix/store/g6qbvfyhp4ragczgqbcg42r38dfqr170-bash-5.1-p16/bin/bash",["-e","/nix/store/9krlzvny65gdc8s7kpb6lkx8cd02c25b-default-builder.sh"],[("LANG","C.UTF-8"),("buildInputs",""),("builder","/nix/store/g6qbvfyhp4ragczgqbcg42r38dfqr170-bash-5.1-p16/bin/bash"),("configureFlags",""),("depsBuildBuild",""),("depsBuildBuildPropagated",""),("depsBuildTarget",""),("depsBuildTargetPropagated",""),("depsHostHost",""),("depsHostHostPropagated",""),("depsTargetTarget",""),("depsTargetTargetPropagated",""),("disabledTests","test_names test_dt_accessor_datetime_name_accessors test_datetime_name_accessors io util_in_top_level test_missing_required_dependency test_from_coo test_comparison_invalid python-kwargs2"),("disallowedReferences",""),("doCheck",""),("doInstallCheck","1"),("hardeningDisable",""),("name","python3.9-pandas-1.4.1"),("nativeBuildInputs","/nix/store/3fyr852wb555h79p92msb2sib54wq196-python3-3.9.11 /nix/store/5cpybwp36mqln3bhbqqmqk54lnikl7ff-hook /nix/store/9gg3hfp4i6ry62k0zw40m00g5q297m9d-hook /nix/store/8yx88xs2wdy5dirdvnknvhi9cv7yw25k-python-remove-tests-dir-hook /nix/store/icgl3zzc0l82qyhxsnnj1y89m1r24vba-python3.9-setuptools-61.2.0 /nix/store/1bfgqv41fbb7d60pys784ny4mai4jhi1-python-catch-conflicts-hook /nix/store/0jz5v40c8psbv97znvzdhqvh84rxd03r-python-remove-bin-bytecode-hook /nix/store/472v23fivaa73p8mdxvwn6c9280jffa2-setuptools-setup-hook /nix/store/lgkdicb0ap4drmxnb91yvyna32wyrjnv-pip-install-hook /nix/store/b7m2x972d81fchpg00w2szgpq01xny2j-python-imports-check-hook.sh /nix/store/mdfwh8and2hwrr1xdzdxx091wwfhcjlc-python-namespaces-hook.sh /nix/store/wb4v9l0w0ah166vmj0apvyzzdhv4yr2b-python3.9-Cython-0.29.28 /nix/store/14v5iha3r8d0xsjalhxs7wcqmjgbhc9z-setuptools-check-hook /nix/store/w2bvq4ajw8f0a15wrlgj3f97za2iz7dm-glibc-locales-2.34-115 /nix/store/kw1mj9la8zbxna740ga08cgbw562600k-python3.9-hypothesis-6.40.0 /nix/store/59jpm1wa0kxk652j3n2lpxmacgp8f2r1-python3.9-Jinja2-3.1.1 /nix/store/xwj3kwa250ndrqhrcwlw9knrvsh730i0-python3.9-pytest-asyncio-0.18.1 /nix/store/vq916azdj3216rk7dahvv8hs56kdvlqy-python3.9-pytest-xdist-2.5.0 /nix/store/7ph1fh2zg40ld9izmf15qz0b75yarazf-pytest-check-hook /nix/store/mrxv9vqynjwy0c6myz4vwsq97jclhvcq-python3.9-xlsxwriter-3.0.2"),("out","/nix/store/5534m7zgrb8jj177aw37cg3na48f4j4a-python3.9-pandas-1.4.1"),("outputs","out"),("patches",""),("pname","pandas"),("postFixup","wrapPythonPrograms\n"),("postPatch",""),("preCheck","cd $out/lib/python3.9/site-packages/pandas\nexport LC_ALL=\"en_US.UTF-8\"\nPYTHONPATH=$out/lib/python3.9/site-packages:$PYTHONPATH\n"),("propagatedBuildInputs","/nix/store/lzrid8xnarrh52ar0rs9yzklbdjvlvdj-python3.9-numpy-1.21.5 /nix/store/31x4b90f9y15fa9him5rv8jkqca7k3g2-python3.9-python-dateutil-2.8.2 /nix/store/kvz5khvfn1a4ra4f0f3lrzv5xmx3z79b-python3.9-pytz-2022.1 /nix/store/3fyr852wb555h79p92msb2sib54wq196-python3-3.9.11"),("propagatedNativeBuildInputs",""),("pytestFlagsArray","--skip-slow --skip-network --numprocesses 0"),("pythonImportsCheck","pandas"),("src","/nix/store/kakz4gcdks9bc4h8c4k1j28shyrj6cgq-pandas-1.4.1.tar.gz"),("stdenv","/nix/store/0i2ipl29khv55c5g8rar91ijc76dsmrz-stdenv-linux"),("strictDeps","1"),("system","x86_64-linux"),("version","1.4.1")])% 
```

--- a/pkgs/development/tools/misc/nix-bisect/default.nix
+++ b/pkgs/development/tools/misc/nix-bisect/default.nix
@@ -1,4 +1,5 @@
 { lib
+, fetchpatch
 , fetchFromGitHub
 , python3
 }:
@@ -17,6 +18,14 @@ python3.pkgs.buildPythonApplication {
     rev = "v${version}";
     hash = "sha256-01vj35mMakqKi5zbMIPQ+R8xdkOWbzpnigd3/SU+svw=";
   };
+
+  patches = [
+    (fetchpatch {
+      # Fixes compatibility with recent nix versions
+      url = "https://github.com/timokau/nix-bisect/commit/01eefe174b740cb90e48b06d67d5582d51786b96.patch";
+      hash = "sha256-Gls/NtHH7LujdEgLbcIRZ12KsJDrasXIMcHeeBVns4A=";
+    })
+  ];
 
   propagatedBuildInputs = with python3.pkgs; [
     appdirs

